solr/core/src/test/org/apache/solr/search/TestFiltering.java

```diff
@@ -86,21 +86,38 @@ public void testLiveDocsSharing() throws Exception {
         QueryResult res = new QueryResult();
         searcher.search(res, cmd);
         set = res.getDocSet();
-        assertTrue( set == live );
+        assertTrue( equals(live, set) );
 
         cmd.setQuery( QParser.getParser(qstr + " OR id:0", null, req).getQuery() );
         cmd.setFilterList( QParser.getParser(qstr + " OR id:1", null, req).getQuery() );
         res = new QueryResult();
         searcher.search(res, cmd);
         set = res.getDocSet();
-        assertTrue( set == live );
+        assertTrue( equals(live, set) );
       }
 
     } finally {
       req.close();
     }
   }
 
+  boolean equals(DocSet ds1, DocSet ds2) {
+    DocSet smaller = ds1.getFixedBitSet().length() < ds2.getFixedBitSet().length() ? ds1: ds2;
```

Comment:
Do we really need this much code here to check this?  I think I've seen DocSet equality elsewhere; perhaps DocSet should implement this for real.  I had taken this away recently but it could return.